telegram/ext/conversationhandler.py

```diff
@@ -455,21 +507,14 @@ def check_update(self, update: object) -> CheckUpdateType:  # pylint: disable=R0
         if isinstance(state, tuple) and len(state) == 2 and isinstance(state[1], Promise):
             self.logger.debug('waiting for promise...')
 
-            old_state, new_state = state
-            if new_state.done.wait(0):
-                try:
-                    res = new_state.result(0)
-                    res = res if res is not None else old_state
-                except Exception as exc:
-                    self.logger.exception("Promise function raised exception")
-                    self.logger.exception("%s", exc)
-                    res = old_state
-                finally:
-                    if res is None and old_state is None:
-                        res = self.END
-                    self.update_state(res, key)
-                    with self._conversations_lock:
-                        state = self.conversations.get(key)
+            # check if promise is finished or not
```

Comment:
IISC we don't strictly need to have that logic in its own method anymore, but I'm fine with leaving it that way.

Reply:
Imo it looks more readable that way, `check_update` is already a very big method.